ssz/sedes/list.py

```diff
@@ -30,6 +30,12 @@
 class List(
     HomogeneousProperCompositeSedes[Sequence[TSerializable], Tuple[TDeserialized, ...]]
 ):
+    def __init__(self, element_sedes: TSedes, max_length: int) -> None:
+        if max_length < 0:
+            raise ValueError(f"Lists must have a size of 0 or greater, got {max_length}")
```

Comment:
I suggest using the following wording for the error message
"Lists must have a **maximum length** of 0 or greater, got {max_length}"